sunpy/map/mapbase.py

```diff
@@ -1482,6 +1484,10 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=4, scale=1.0,
         pad_x = int(np.max((diff[1], 0)))
         pad_y = int(np.max((diff[0], 0)))
 
+        if issubclass(self.data.dtype.type, numbers.Integral):
+            warnings.warn("Integer map data is incompatilbe with specified missing value",
```

Comment:
I think a test to make sure this is raised would be nice.

Reply:
I'm not sure as to what datatype the testing map is, should I have to load a separate map in the test to make sure this works?

Reply:
You can find out by loading the map? Or you can make a fake one with a fixed data type. Both would work.

Reply:
Yeah. I'll get this done.